src/BlockTypes/ProductGalleryThumbnails.php

```diff
@@ -170,13 +170,14 @@ protected function render( $attributes, $content, $block ) {
 					}
 
 					return sprintf(
-						'<div class="wc-block-product-gallery-thumbnails wp-block-woocommerce-product-gallery-thumbnails %1$s" style="%2$s" data-wc-interactive=\'%4$s\'>
-							%3$s
+						'<div class="wc-block-product-gallery-thumbnails wp-block-woocommerce-product-gallery-thumbnails %1$s wc-block-product-gallery-thumbnails__total-count-%2$s" style="%3$s" data-wc-interactive=\'%4$s\'>
```

Comment:
I don't think it's necessary to add this additional class for tracking the number of thumbnails.

You can utilize the attribute that's on the parent Product Gallery element.

```
		#{$gallery}[data-thumbnails-number-of-thumbnails='4'] & {

		}
```

Note: this class is not present when the number of thumbnails is default, that is `3`.